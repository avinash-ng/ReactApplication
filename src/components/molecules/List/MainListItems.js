import React from 'react' 
import { ListItem, ListItemText } from '@material-ui/core' 
const MainListItems = (props) => {
    var menus = ['Home', 'About', 'Contact', 'Gallery']
    return (
        
            menus.map((text, index) => (
                <ListItem button style={{ marginLeft: '0px' }} key={text} onClick={() => props.clicked(text)}>
                    <ListItemText primary={text} />
                </ListItem>
            ))
        )
}
export default MainListItems