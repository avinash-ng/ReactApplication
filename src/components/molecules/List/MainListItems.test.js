import MainListItems from './MainListItems'
import expect from 'expect'
import sinon from 'sinon';
import { shallow, configure, mount } from 'enzyme'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import { render, cleanup, fireEvent } from "@testing-library/react";
configure({ adapter: new Adapter() });
describe('<MainListItems /> ', () => {
    it('renders list item < home />', () => {

        const {getByText} = render(
            <MainListItems ></MainListItems>
        );
        expect(getByText('Home')).toBeInTheDocument()
        // console.log(RenderComponent)
        
    })

    it('renders list item <About />', () => {

        const { getByText } = render(
            <MainListItems ></MainListItems>
        );
        expect(getByText('About')).toBeInTheDocument()
        // console.log(RenderComponent)

    })

    it('checking click functionality', () => {
        const clicked = sinon.spy()
        const { getByText } = render(
            <MainListItems clicked={clicked}> </MainListItems>
        )
        fireEvent.click(getByText('Home'))
        expect(clicked).toHaveProperty('callCount', 1)
    })

    it('Matches snapshot', () => {
        const clicked = sinon.spy()
        const RenderComponent = shallow(<MainListItems clicked={clicked}> </MainListItems>)
        expect(RenderComponent).toMatchSnapshot()
    })
})