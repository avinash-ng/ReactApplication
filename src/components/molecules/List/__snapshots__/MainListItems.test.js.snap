// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainListItems />  Matches snapshot 1`] = `ShallowWrapper {}`;
